DEVELOPMENT.md

```diff
@@ -87,6 +87,16 @@ Maintain the same quality for production and test code.
 
 When writing a Git commit message, follow these [guidelines](https://chris.beams.io/posts/git-commit/).
 
+### Git merge strategy
+
+We prefer to `rebase and merge`, so cleaning up and squashing commits on the feature branch is best.
+`squash and merge` is reserved for when git is unfamiliar to the external contributor. 
+We care a lot about the history, so this is why how change is structured between commits and what the
+commit messages are is also a part of code review. However we prefer not to merge anything where changes goes
+back and forth between commits in the same PR. This makes the entire git history much cleaner. Notice that
+writing code is creative process and it is natural to got back and forth when writing the code, however
+when visiting history such small steps are not helpful.
```

Comment:
In general content is ok to me, but it requires some language improvements from native speakers.

@mosabua or @electrum Would you like to help here?